packages/base/build.yaml

```diff
@@ -1,15 +1,31 @@
 image: "{{.Values.image}}"
 
+env:
+{{ if .Values.distribution }}
+{{if eq .Values.distribution "ubuntu" }}
+- DEBIAN_FRONTEND=noninteractive
+{{end}}
+{{end}}
 steps:
-{{if .Values.is_opensuse}}
-- zypper in -y --no-recommends {{.Values.leap_packages}} {{.Values.common_packages}} {{.Values.kernel_package}}
+{{ if .Values.distribution }}
+{{if eq .Values.distribution "opensuse" }}
+- zypper in -y --no-recommends {{.Values.packages}} {{.Values.kernel_package}}
 - zypper cc
-{{else if .Values.is_fedora}}
+{{else if eq .Values.distribution "fedora" }}
 - echo "install_weak_deps=False" >> /etc/dnf/dnf.conf
-- dnf install -y {{.Values.fedora_packages}} {{.Values.common_packages}} {{.Values.kernel_package}}
+- dnf install -y {{.Values.packages}} {{.Values.kernel_package}}
 - dnf clean all
 - |
     kernel=$(ls /boot/vmlinuz-* | head -n1) && \
     ln -sf "${kernel#/boot/}" /boot/vmlinuz
+{{else if eq .Values.distribution "ubuntu" }}
+# Fun: https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/1879558
+- apt-get update && apt-get install -y {{.Values.packages}} {{.Values.kernel_package}} || true
```

Comment:
I'm trying to see if I can get rid of the `|| true` here now. At that time I was hitting https://bugs.launchpad.net/ubuntu/+source/grub2/+bug/1879558 , and post-install hooks were failing due to both grub-386 and amd64 missing

Reply:
I believe I already went through this issue. Grub debian packages are so broken and it's deb sepc is just insane. At time I realized one option was to split grub packages installation into two transactions (bios boot first and then efi boot on a second round). Alternatively, for Focal version I found that installing this exact list of grub packages was not failing in a single transaction and was pulling the appropriate dependencies.

```
grub-pc-bin
grub2-common
grub-efi-amd64-bin
```

So from your case I would drop `grub2` package from the list (which is already duplicated by the way)